actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/CourseBatchNotificationActor.java

```diff
@@ -0,0 +1,421 @@
+      }
+    } else {
+      ProjectLogger.log(
+          "BatchOperationNotifierActor : sendMail, Error encountered while sending email, eitheer it is black or null ",
+          LoggerEnum.ERROR.name());
+    }
+  }
+
+  /*
+   * This method takes CourseBatch object and returns list of participants
+   *
+   * @param CourseBatch object
+   *
+   * @returns list of users
+   */
+  private List<String> getParticipants(CourseBatch courseBatch) {
+    List<String> usersIds = null;
+    Map<String, Boolean> participants = courseBatch.getParticipant();
+    if (participants != null) {
+      usersIds = new ArrayList<>();
+      Set<String> keys = participants.keySet();
+      for (String user : keys) {
+        if (participants.get(user)) {
+          usersIds.add(user);
+        }
+      }
+    }
+    return usersIds;
+  }
+
+  /*
+   * This method takes List<String>userIds and returns List<Map<User,Objects>> of
+   * participants
+   *
+   * @param List<String> userIds list of userIds
+   *
+   * @return List<Map<String, Object>> map of user related data.
+   */
+  private List<Map<String, Object>> getUsersFromDB(List<String> userIds) {
+    Util.DbInfo usrDbInfo = Util.dbInfoMap.get(JsonKey.USER_DB);
+    List<String> userIdList = new ArrayList<>(userIds);
+    List<String> fields = new ArrayList<>();
+    fields.add(JsonKey.FIRST_NAME);
+    fields.add(JsonKey.EMAIL);
+    Response response =
+        cassandraOperation.getRecordsByIdsWithSpecifiedColumns(
+            usrDbInfo.getKeySpace(), usrDbInfo.getTableName(), fields, userIdList);
+    if (response == null) {
+      ProjectLogger.log("No data from cassandra , check connection  ", LoggerEnum.ERROR.name());
+    }
+    List<Map<String, Object>> userList = (List<Map<String, Object>>) response.get(JsonKey.RESPONSE);
+    return userList;
+  }
+
+  /*
+   * This method takes template name and return template.
+   *
+   * @param templateName String
+   *
+   * @return email template String .
+   */
+  private String getEmailTemplateFile(String templateName) {
+    EmailTemplateDao emailTemplateDao = EmailTemplateDaoImpl.getInstance();
+    String template = emailTemplateDao.getTemplate(templateName);
+    if (StringUtils.isBlank(template)) {
+      ProjectCommonException.throwClientErrorException(
+          ResponseCode.invalidParameterValue,
+          MessageFormat.format(
+              ResponseCode.invalidParameterValue.getErrorMessage(),
+              templateName,
+              JsonKey.EMAIL_TEMPLATE_TYPE));
+    }
+    return template;
+  }
+
+  public static VelocityContext getContext(Map<String, Object> map) {
+    VelocityContext context = new VelocityContext();
+    if (StringUtils.isNotBlank((String) map.get(JsonKey.COURSE_NAME))) {
+      context.put(JsonKey.COURSE_NAME, getValue(map, JsonKey.COURSE_NAME));
+    }
+    if (StringUtils.isNotBlank((String) map.get(JsonKey.START_DATE))) {
+      context.put(JsonKey.BATCH_START_DATE, getValue(map, JsonKey.START_DATE));
```

Comment:
Why don't we use existing constants (START_DATE and END_DATE) instead of creating new constants (BATCH_START_DATE and BATCH_END_DATE)?

Reply:
for setting it in template , as template have this place holders